bin/comment_on_pr

```diff
@@ -1,14 +1,146 @@
 #!/bin/bash -eu
 
-PR_MESSAGE_FILE=$1
-GITHUB_TOKEN=$2
+# This script is used to comment on a pull request, and must be run in a buldkite PR step.
+#
+# Usage:
+#    [GITHUB_TOKEN=<token>] comment_on_pr [--id <comment-id>] [--if-exist update|delete] [--github-token <token>] <comment>
+#
+# Examples:
+#
+#    # Post a new comment which can be uniquely identified by the given comment id,
+#    # or update the existing comment that's associated with the given comment id.
+#    # By default, the existing comment will be updated. You can choose to delete
+#    # the existing comment by passing `--if-exist delete`.
+#    comment_on_pr --id comment-id [--if-exist update|delete] "This is a comment"
+#
+#    # Delete the comment associated with the given comment id by passing an empty comment
+#    comment_on_pr --id comment-id [--if-exist delete]
+#
+#    # Deprecated: Post a new comment to the PR
+#    comment_on_pr "This is a comment" <github-token>
+#
+# Please note the comment argument can be a path to a file containing the comment, or a markdown.
 
-if ! command -v gh &> /dev/null; then
-	brew install gh
+# Check dependencies and print their versions for diagnosis purposes
+cat <<EOF
+> jq --version
+$(jq --version)
+
+> curl --version
+$(curl --version)
+
+EOF
+
+if [[ ! "${BUILDKITE_PULL_REQUEST:-invalid}" =~ ^[0-9]+$ ]]; then
+    echo "Error: this tool can only be called from a Buildkite PR job"
+    exit 1
+fi
+
+# Default options
+opt_comment_id=""
+opt_if_exist="update"
+opt_github_token="${GITHUB_TOKEN:-}"
+
+# If there are two arguments and they don't start with an option, then we assume
+# this utility is called in the deprecated way: comment_on_pr <comment> <github-token>
+if [[ $# == 2 && "$1" != "--"* ]]; then
+    if [[ -f "$1" ]]; then
+        arg_pr_comment=$(cat "$1")
+    else
+        arg_pr_comment="$1"
+    fi
+
+    opt_github_token=$2
+else
+    while [[ "$#" -gt 0 ]]; do
+        case $1 in
+            --id) opt_comment_id="$2"; shift ;;
+            --if-exist) opt_if_exist="$2"; shift ;;
+            --github-token) opt_github_token="$2"; shift ;;
+            --*) echo "Unknown option: $1"; exit 1 ;;
+            *) break ;;
+        esac
+        shift
+    done
+
+    case "$#" in
+        0)
+            arg_pr_comment=""
+            ;;
+        1)
+            if [[ -f "$1" ]]; then
+                arg_pr_comment=$(cat "$1")
+            else
+                arg_pr_comment="$1"
+            fi
+            ;;
+        *) echo "Error: too many arguments"; exit 1 ;;
+    esac
+fi
+
+export GITHUB_TOKEN="$opt_github_token"
+
+github_user=$(basename "$(dirname "${BUILDKITE_PULL_REQUEST_REPO%.git}")")
```

Comment:
💄  nit: `%.git` is not really necessary for this case because we're taking the `dirname` anyway so don't care if the basename ends in `.git`. Thus that `%.git` is only necessary for the `github_repo` computation below, but not needed for the `github_user` 🙃

Reply:
Good catch! Addressed in https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/51/commits/250e38c97aa6591693610f978473998f16af1848